src/test/java/org/cactoos/text/PaddedTextStartTest.java

```diff
@@ -0,0 +1,83 @@
+/**
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2017 Yegor Bugayenko
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.cactoos.text;
+
+import java.io.IOException;
+import org.hamcrest.MatcherAssert;
+import org.cactoos.TextHasString;
+import org.cactoos.text.NotNullText;
+import org.junit.Test;
+
+/**
+ * Test case for {@link PaddedTextStart}.
+ *
+ * @author Vivek Poddar (vivekimsit@gmail.com)
+ * @version $Id$
+ * @since 0.1
+ * @checkstyle JavadocMethodCheck (500 lines)
+ */
+public final class PaddedTextStartTest {
+
+    @Test
+    public void noPadding() {
+        MatcherAssert.assertThat(
+            "Shouldn't pad the text",
+            new PaddedTextStart(
+              new NotNullText(
+                new StringAsText("x")
+              ), 1, '-'),
+            new TextHasString("x")
+        );
+    }
+
+    @Test
+    public void somePadding() {
+        MatcherAssert.assertThat(
+            "Should pad chars at start",
+            new PaddedTextStart(
+              new NotNullText(
+                new StringAsText("x")
+              ), 2, '-'),
+            new TextHasString("-x")
+        );
+    }
+
+    @Test
+    public void negativeMinLength()  {
+        MatcherAssert.assertThat(
+            "Shouldn't consider negative min length",
+            new PaddedTextStart(
+              new NotNullText(
+                new StringAsText("x")
+              ), -1, '-'),
+            new TextHasString("x")
+        );
+    }
+
+    @Test(expected = IOException.class)
+    public void failForNullText() throws IOException {
```

Comment:
Since, a unit test serves as a better documentation hence, the user should know how an object behaves in different scenarios. How would he know that my object doesn't handle `Null` text?